src/middleware/builtin.ts

```diff
@@ -91,7 +91,7 @@ export function matchConstraints(
         }
       } else {
         if ((tempMatches = constraints.block_id.exec(payload.block_id)) !== null) {
```

Comment:
out of curiosity, aren't these `exec()` calls affected in the same way? should they also turn into `while` loops?

Reply:
I made an assumption that developers wouldn't use multiple matches in constraint objects, but maybe we shouldn't make that assumption?